### ISSUE TITLE
Move sorting logic into Table component

### ISSUE THREAD

**Issue by timroes** [airbytehq/airbyte#24426]
Currently we handle sorting the data outside the `Table` component. That leads to problems, since we need to handle sorting (often with the very same code) in every consuming place. Also we need to manually pass in the `sortedByColumn` property again to actually highlight the sorted column. Since React Table is doing some "magic" transforming of the accessor key (i.e. the accessor might be `connection.connectionName`, but the column ID will be `connection_connectionName` we even need to mirror that behavior to set the sorting key).

tl;dr: we should move sorting into the `Table` component. React Table already has support for setting sorting columns etc, that we can leverage for this.

**Comment by tealjulia** [comment]
We can also do some work in `useCreditsUsage` to unnest the properties used in that table.  I think that's the only table that does the weird nested property thing.

**Comment by josephkmh** [comment]
This is something we want to do in Q2, but has lower priority than the `ListBox` or `react-hook-form` work

**Comment by timroes** [comment]
Closed as of https://github.com/airbytehq/airbyte-platform-internal/pull/8395
